src/main/java/net/minecraftforge/common/conditions/NotCondition.java

```diff
@@ -5,58 +5,26 @@
 
 package net.minecraftforge.common.conditions;
 
-import com.google.gson.JsonObject;
-import net.minecraft.util.GsonHelper;
-import net.minecraft.resources.ResourceLocation;
+import com.mojang.serialization.Codec;
+import net.minecraftforge.common.ForgeMod;
 
-public class NotCondition implements ICondition
+public record NotCondition(ICondition value) implements ICondition
 {
-    private static final ResourceLocation NAME = new ResourceLocation("forge", "not");
-    private final ICondition child;
-
-    public NotCondition(ICondition child)
-    {
-        this.child = child;
-    }
-
-    @Override
-    public ResourceLocation getSerializerId()
-    {
-        return NAME;
-    }
-
     @Override
     public boolean test(IContext context)
     {
-        return !child.test(context);
+        return !value.test(context);
     }
 
     @Override
     public String toString()
     {
-        return "!" + child;
+        return "!" + value;
```

Comment:
This should be modified to add parentheses to the output, since it would be misleading to read a NOTed AND/OR condition (`!test1 && test2` vs `!(test1 && test2)`).